### PR TITLE
Add option to auto-orient normals of `contour_labels` filter

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Split from #6732.

The default orientation of the cells generated by this filter is inconsistent, such that the external surface often points inwards instead of out. I think it's desirable to auto-orient the generated contours by default so that the normals point outward.

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by tkoyama010** [comment]
It has not yet been released, so I removed a breaking-change label.

File: pyvista/core/filters/image_data.py [hunk 1]
```diff
@@ -1103,6 +1104,22 @@ def contour_labels(  # type: ignore[misc]
...
+            Compute point and cell normals for the contoured output using
+            :meth:`~pyvista.PolyDataFilters.compute_normals` with ``auto_orient_normals``
+            enabled by default. If ``False``, the generated polygons may have
+            inconsistent ordering and orientation (and may negatively impact
+            the shading used for rendering).
+
+            .. warning::
+
+                Enabling this option is likely to generate surfaces with normals
+                pointing outward when ``closed_surface`` is ``True`` and
```
Comment (user27182) [inline pyvista/core/filters/image_data.py:1117]: Whoops, this term is left over from before #7028 and should be updated
``` diff
- ``closed_surface``
+ ``pad_background``
```

Comment (user27182) [reply pyvista/core/filters/image_data.py:1117]: This change is captured in #7039

**Pull request merged**